src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private DtoResponseMapper<UserResponseDto, User> responseMapper;
+    private UserService userService;
+
+    public UserController(DtoResponseMapper<UserResponseDto, User> responseMapper,
+                          UserService userService) {
+        this.responseMapper = responseMapper;
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getUserByEmail(@RequestParam String email) {
+        return responseMapper.toDto(userService.findByEmail(email).orElseThrow(
+                () -> new DataProcessingException("User with email: "
```

Comment:
maybe you should change the type of exception to "NoSuchElement" cuz this is controller level, not dao